makelib/common.mk

```diff
@@ -106,6 +106,10 @@ HOSTARCH := $(shell uname -m)
 ifeq ($(HOSTARCH),x86_64)
 HOSTARCH := amd64
 endif
+# For Apple Silicon we are using amd64 binaries and relying on Rosetta2
```

Comment:
Is this because we don't have the automation to build & push it to the registries?

Reply:
This change was initiated because upbound/local-dev-config relies on this submodule to bootstrap a local development environment.

In makelib/local.mk we are downloading binaries based on the architecture defined in this file. Currently kubernetes-cli, helm, and several other tools do not publish darwin_arm64 binaries. But the darwin_amd64 binaries run on the Apple M1, via Rosetta2.

So until upstream binaries are made available, we are defaulting to running the darwin_amd64 binaries.

Reply:
Got it. So this is more about the tools to download rather than how to compile the code in the current repository. Thanks for the explanation!